plugins/elanfp/fu-elanfp-file-control.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2021
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "fu-elanfp-file-control.h"
+
+#define SIZE_IDENTIFY_PACKET 16
+
+#define S2F_TAG_FIRMWAREVERSION 0x00
+#define S2F_TAG_CFU_OFFER_A	0x72
+#define S2F_TAG_CFU_OFFER_B	0x73
+#define S2F_TAG_CFU_PAYLOAD_A	0x74
+#define S2F_TAG_CFU_PAYLOAD_B	0x75
+#define S2F_TAG_END_OF_INDEX	0xFF
+
+gboolean
+binaryVerify(const guint8 *pbinary, gsize binary_size, S2FFILE *ps2ffile, GError **error)
+{
+	S2FINDEX *ps2findex = NULL;
+	guint8 *pindex = NULL;
+	gboolean result = TRUE;
+
+	if (pbinary == NULL)
+		return FALSE;
+
+	ps2ffile->S2FHeader.Tag = *(guint32 *)pbinary;
+	ps2ffile->S2FHeader.FormatVersion = *(guint32 *)(pbinary + 4);
+	ps2ffile->S2FHeader.ICID = *(guint32 *)(pbinary + 8);
+	ps2ffile->S2FHeader.Reserve = *(guint32 *)(pbinary + 12);
```

Comment:
this is rather scary (e.g. what happens if the binary size is 1 byte?) and not at all endian safe. I think you need to use fu_common_read_uint32_safe() here

Reply:
Fixed in commit 5af9c23594ac261d85fbf53fca02de08a5ef0150.